src/frontend/web_application/src/components/MessageList/components/Message/presenter.jsx

```diff
@@ -86,51 +62,78 @@ class Message extends Component {
     }
   }
 
-  setBodyHeight() {
-    const message = this.props.message;
-    const body = message.body;
-    if (body.length > 140) {
-      this.setState({
-        isTooLong: true,
-        isExpanded: false,
-        body: body.substring(0, 140),
-      });
-    } else {
-      this.setState({
-        body,
-      });
-    }
-  }
+  setContentHeight = () => {
+    const bodyHeight = this.divElement.clientHeight;
 
-  handleActiveClick() {
     this.setState(prevState => ({
-      isActive: !prevState.isActive,
+      ...prevState,
+      bodyHeight,
+      isTooLong: bodyHeight > FOLD_HEIGHT && true,
+      isFold: bodyHeight > FOLD_HEIGHT && !prevState.isUnread && true,
     }));
   }
 
-  handleExpandClick() {
-    const message = this.props.message;
-    const body = message.body;
+  handleExpandClick = () => {
+    const bodyHeight = this.divElement.clientHeight;
     this.setState(prevState => ({
-      isExpanded: !prevState.isExpanded,
-      body: prevState.isExpanded ? body.substring(0, 140) : body,
+      isFold: !prevState.isFold,
+      bodyHeight,
     }));
   }
 
+  renderMessageContent = () => {
+    const { message } = this.props;
+
+    const contentProps = {
+      className: classnames(
+        'm-message__content',
+        { 'm-message__content--fold': this.state.isFold },
+      ),
+      style: {
+        height: this.state.isFold ? null : this.state.bodyHeight,
+        transitionTimingFunction: this.state.isFold ? 'ease-in-out' : 'ease-out', // to make 'expand'/'collapse' animation smoother
+        transitionDuration: `${((this.state.bodyHeight / 100) * 0.05)}s`, // to make 'expand'/'collapse' animation smoother
+      },
+    };
+
+    const bodyProps = {
+      className: classnames(
+        'm-message__body',
+        { 'm-message__body--rich-text': !message.body_is_plain },
+      ),
+      ref: (divElement) => { this.divElement = divElement; },
+    };
+
+    return (
+      <div {...contentProps}>
+        {!message.body_is_plain ? (
+          <div {...bodyProps} dangerouslySetInnerHTML={{ __html: message.body }} />
+        ) : (
+          <pre {...bodyProps}>{message.body}</pre>
+        )
+        }
+      </div>
+    );
+  }
+
   render() {
     const { message, locale, onDelete, __ } = this.props;
     const author = message.participants.find(participant => participant.type === 'From');
     const subject = message.subject;
+    const dropdownId = uuidV1();
+    const typeTranslations = {
+      email: __('message-list.message.protocol.email'),
+    };
+
     const topBarClassName = classnames(
       'm-message__top-bar',
-      { 'm-message__top-bar--active': this.state.isActive }
-    );
-    const bodyClassName = classnames(
-      'm-message__body__content',
-      { 'm-message__body__content--expanded': this.state.isExpanded }
+      { 'm-message__top-bar--is-unread': this.state.isUnread },
```

Comment:
I don't understand why this does not work `message.is_unread`

Reply:
btw the modifier could be on the Block: `m-message--is-unread__top-bar` (same for the next class: `m-message--is-unread__subject`)